fastchat/serve/openai_api_server.py

```diff
@@ -51,92 +61,106 @@ class AppSettings(BaseSettings):
 app = fastapi.FastAPI()
 headers = {"User-Agent": "FastChat API Server"}
 
+def create_error_response(code: int, message: str) -> JSONResponse:
+    return JSONResponse(ErrorResponse(message=message, code=code).dict(), status_code=500) 
 
-@app.get("/v1/models")
-async def show_available_models():
-    controller_address = app_settings.controller_address
-    async with httpx.AsyncClient() as client:
-        ret = await client.post(controller_address + "/refresh_all_workers")
-        ret = await client.post(controller_address + "/list_models")
-    models = ret.json()["models"]
-    models.sort()
-    return {"data": [{"id": m, "object": "model"} for m in models],
-            "object": "list"}
-
-
-@app.post("/v1/chat/completions")
-async def create_chat_completion(request: ChatCompletionRequest):
-    """Creates a completion for the chat message"""
-    gen_params = get_gen_params(
-        request.model,
-        request.messages,
-        temperature=request.temperature,
-        max_tokens=request.max_tokens,
-        echo=False,
-        stop=request.stop,
-    )
-    if request.stream:
-        response_stream = chat_completion_stream(
-            request.model, gen_params, request.n)
-        return StreamingResponse(response_stream, media_type="text/event-stream")
 
-    # TODO: batch the requests
-    choices = []
-    chat_completions = []
-    for i in range(request.n):
-        content = asyncio.create_task(chat_completion(request.model, gen_params))
-        chat_completions.append(content)
+@app.exception_handler(RequestValidationError)
+async def validation_exception_handler(request, exc):
+    return create_error_response(50001, str(exc))
```

Comment:
could you use global constants for error codes? same below

Reply:
I added a global Error Code class to this file now. I think this error code may be needed elsewhere, so I am finding a better place to list these error codes.